### ISSUE TITLE
WORK IN PROGRESS: Finalize can-stache-animate

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache-animate#46]
__tldr;__ This proposal seeks to finish a safe-as-possible working "core" set of functionality for `can-stache-animate`.  Critically, this proposal doesn't seek to provide any animations other than a `fadeOut` and `fadeIn` animation which are primarily useful for demonstrating how to build animations.

By _safe-as-possible_ we mean that this doesn't create any unnecessary globals.
By _core_ we mean that there's just enough functionality for people to create their own animation behaviors.

## How animations should work 

Animation functions should run via `can-stache-bindings` event-bindings and look like a function that can be easily imported and passed around:

```js
<can-import from="fade-out" value:to="scope.helpers.fadeOut"/>
<div on:click="fadeOut(100)"/>
```

For events on viewModels, folks should use `:by:` like:

```js
<can-import from="fade-out" value:to="scope.helpers.fadeOut"/>
<div on:updated:by:todo="fadeOut(100)"/>
```
## Chaining suggestions 

<div on:updated:by:todo="fadeOut(100, todo.doSomething)"/>

## Special Mutation Events

It will be useful to support animations that occur in relationship to changes in the DOM. Specifically:

- an event for when an element is inserted into the page (ex: `connected`)
- an event for when an element is removed from the page, where the animation can control when the actual removal takes place  (ex: `beforedisconnect`)

Critically, these methods can __NOT__ be asynchronous when fired.  They need to work as follows (should log 1 then 2):

```js
var domEvents = require("can-dom-events");
var connectedEvent = require("can-dom-events");
domEvents.addEvent(connectedEvent);

var div = document.createElement("div");
domEvents.addEventHandler(div,"connected", function(){
  console.log(1);
});

document.body.appendChild(div);
console.log(2);
```

- making it work for common use cases
- docs
- test

### Into local helpers

```
<can-import from="can-stache-animage" value:to="scope.helpers.animate"/>
<div on:click="animate.fadeOut"/>
```

### Long / safe way

```
<can-import from="can-stache-animate" value:to="scope.vars.animate"/>

<div on:click="scope.vars.animate.fadeOut()"/>
```

Alternate:

```js
import animate from 
Component.extend({
  ViewModel: {
    animate: {default: animate}
  },
 ...
})
```

### Short / unsafe way

- Good for JSBins

```
<can-import from="can-stache-animate"/>

<div on:click="animate.fadeIn()"/>
```
